packages/hardhat/contracts/PredictorGame.sol

```diff
@@ -76,4 +79,25 @@ contract PredictorGame {
         limits[1] = maxBet;
         return limits;
     }
+
+	function _startRound(uint256 epoch) private {
+		Round storage round = rounds[epoch];
+		round.startTimestamp = block.timestamp;
+		round.closeTimestamp = block.timestamp + 5 minutes;
+		round.epoch = epoch;
+		round.totalAmount = 0;
+
+		emit StartRound(epoch);
```

Comment:
should we also consider taking in the price as an argument on this function so it can be added to the StartRound event? 

I'm still working on understand the full picture so maybe we are holding on to the start price some other way

Reply:
A new round is always the next round in the future for which predictions can be made, and there are no prices yet. The price of its opening can be obtained after the closing of the current round.
Here's a textual representation of the three rounds:

Round 1 (lastRound):
Open price: Available
Close price: Available

Round 2 (currentRound):
Open price: Available
Close price: Not available

Round 3 (nextRound):
Open price: Not available
Close price: Not available

Reply:
so the plan is to have chainlink automation call the `_startRound` function every 5 minutes 24 hours a day? so most of the rounds will not have any players?

Reply:
We need to create a function called `manageRound` that will close the current round once and start the next one.

The game runs round the clock, designed to cover all time zones, much like trading. Currently, there are three main sessions, each lasting 8 hours: America, Europe, and Asia. We need to develop bots that can autonomously place small bets to simulate activity. Even for the game demonstration, these bots will be necessary.

Reply:
Can you help me understand the advantage of automated round starts that cost a bunch more overhead vs only starting a round when a user places a bet?

Reply:
I think similar to @MattPereira, rounds could be triggered to start if at least one user enters, using Chainlink Automation to verify if the condition has been met before triggering a transaction.